### ISSUE TITLE
Expose Invalid Subpattern Symbol in error cases

### ISSUE THREAD

**Issue by 333fred** [dotnet/roslyn#33175]
We currently don't expose invalid `ISymbol`s on an `IPropertySubpattern.Member` if the symbol is something other than `IFieldSymbol`, `IPropertySymbol`, or `null`. We need to expose this is in some manner.

**Comment by 333fred** [comment]
We also need to update the CFG to synthesize the correct receivers for PropertySubpatterns, taking into account nested subpatterns. For a nested example, see `IsPattern_NoControlFlow_03`. The first level pattern should have the `IInstanceReceiverOperation` replaced with a capture reference to the input, and then nested `IInstanceRecevierOperation`s should have synthesized property accesses off that initial receiver.

**Comment by 333fred** [comment]
We also need to revert https://github.com/dotnet/roslyn/pull/33178/files#r254423185 when the interface is made public.

**Comment by 333fred** [comment]
https://github.com/dotnet/roslyn/pull/33178#discussion_r254487600

**Comment by 333fred** [comment]
Comments from Chuck:
https://github.com/dotnet/roslyn/pull/33178#discussion_r254495124
https://github.com/dotnet/roslyn/pull/33178#discussion_r254495124
https://github.com/dotnet/roslyn/pull/33178#discussion_r254498066
https://github.com/dotnet/roslyn/pull/33178#discussion_r254500027
https://github.com/dotnet/roslyn/pull/33178#discussion_r254500360

**Comment by gafter** [comment]
> We also need to update the CFG to synthesize the correct receivers for PropertySubpatterns, taking into account nested subpatterns. For a nested example, see `IsPattern_NoControlFlow_03`. The first level pattern should have the `IInstanceReceiverOperation` replaced with a capture reference to the input, and then nested `IInstanceRecevierOperation`s should have synthesized property accesses off that initial receiver.

I disagree.  The point of these nodes is to specify which field is being pattern-matched, not to describe a read operation that occurs at any particular time.  Turning the receiver into a capture would require reading it, which is not correct.

**Comment by gafter** [comment]
The comments in the comment stream of this issue have been addressed. The OP remains an open issue.

**Comment by alrz** [comment]
>or null

How could it be null? isn't it an `IInvalidOperation` if neither field or property is available?

**Comment by gafter** [comment]
Yes, we should use an `IInvalidOperation` for `Member` if none exists. I guess I was thinking the type of `IPropertySubpatternOperation.Member` was `ISymbol`, but it is `IOperation`.
